### PR TITLE
Log in view models controllers

### PR DESCRIPTION
**Pull request by FranDepascuali**
## Sumary

**Trello cards**:
tech-login: https://trello.com/c/CyFcx2ll/21-tech-login
login: https://trello.com/c/Uv3haBcZ/14-login

Developed log-in view controllers and models.
## Test Plan
- Open the app
- Login with username 'f@wolox.com.ar' password 'probando'
- You should see a spinner while logging in.
- The spinner should disappear when logged.
- If mail or password aren't correct, a toast with label "error" will appear. 
- If signup button is clicked, it leads to signUp form.
## Screenshots

login
## Known issues
- Errors are not yet classified.
- News controllers and model not yet implemented, successful log in will lead to news controller.

### CODE REVIEW

**Comment by FranDepascuali** [comment]
@pablog

**Comment by FranDepascuali** [comment]
Deleted Basemodel

File: Wolox Tranining App.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -295,8 +307,11 @@
 				B99D46311B00E15600A604D4 /* UserRequestManager.m in Sources */,
 				B9902E361AEFDCD70015B7A8 /* RequestManager.m in Sources */,
 				F8CD9DE51AE7E6C400D4FF53 /* AppDelegate.m in Sources */,
+				B970F9FE1AF7C3C8007468C4 /* BaseModel.m in Sources */,
```
Comment (pablog) [inline Wolox Tranining App.xcodeproj/project.pbxproj]: Por que tenes BaseModel.m aca? No deberia estar este archivo.

File: Wolox Tranining App/SignUpViewController.h [hunk 2]
```diff
@@ -16,4 +16,5 @@
 @property (weak, nonatomic) IBOutlet UIButton *termsButton;
 @property (weak, nonatomic) IBOutlet UIButton *createUserButton;
 
+
```
Comment (pablog) [inline Wolox Tranining App/SignUpViewController.h:18]: Revisa siempre lo que estas subiendo antes de commitear.

File: Wolox Tranining App/LogInViewController.m [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    [super viewDidLoad];
+    self.loadingActivityIndicator.hidden = YES;
+    self.viewModel = [[LogInViewModel alloc] init];
+}
+
+- (IBAction)logInButtonClicked:(id)sender {
+    self.loadingActivityIndicator.hidden = NO;
+    [self.loadingActivityIndicator startAnimating];
+    [self.viewModel logInWithEmail:self.emailTextField.text password:self.passwordTextField.text success:^ {
+        [self.loadingActivityIndicator stopAnimating];
```
Comment (pablog) [inline Wolox Tranining App/LogInViewController.m]: El activity indicator tiene una property que es `hidesWhenStopped` que te evita tener que hacer el `hidden` a mano.

File: Wolox Tranining App/LogInViewController.m [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+@implementation LogInViewController
+
+- (IBAction)termsButtonClicked:(id)sender {
+    [self.viewModel openTermsAndConditions];
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    self.loadingActivityIndicator.hidden = YES;
```
Comment (pablog) [inline Wolox Tranining App/LogInViewController.m]: Aca tambien podes usar la property del hide.

File: Wolox Tranining App/LogInViewModel.m [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+@property (strong, nonatomic) UserRequestManager *manager;
+
+@end
+
+@implementation LogInViewModel
+
+- (id)init {
+    self = [super init];
+    self.manager = [[UserRequestManager alloc] init];
```
Comment (pablog) [inline Wolox Tranining App/LogInViewModel.m]: if self.

**Comment by FranDepascuali** [comment]
@pablog

**Comment by pablog** [comment]
:+1:

**Pull request closed without merging**